lib/fprotect/Kconfig

```
@@ -50,3 +50,10 @@ config FPROTECT_BLOCK_SIZE
 	default $(dt_node_int_prop_hex,$(DT_CHOSEN_ZEPHYR_FLASH),erase-block-size)
 	help
 	  Block size of fprotect
+
+config FPROTECT_ENABLE_NO_ACCESS
+	bool "Enable no access protection for flash regions"
+	depends on HAS_HW_NRF_ACL
```

Does this need to be configurable?
Alternatively, it could be renamed to FPROTECT_HAS_NO_ACCESS, and always be y when you have ACL, and n if not.
```suggestion
config FPROTECT_HAS_NO_ACCESS
	bool
	default y if HAS_HW_NRF_ACL
```

---

Hm, Fprotect is not enabled by default I think, It's a configurable option. We can enabled that when fprotect is enabled and ACL hardware is there though indeed.

---

I didn't notice it wasn't inside an `if FPROTECT` block. With the `depends on` I added below, it will only be enabled if FPROTECT is enabled.

```suggestion
config FPROTECT_HAS_NO_ACCESS
	bool
	default y if HAS_HW_NRF_ACL
	depends on FPROTECT
```